mParticle-Apple-SDK/mParticle.m

```diff
@@ -159,6 +159,89 @@ - (NSString *)description {
 @implementation MPDataPlanOptions
 @end
 
+@implementation MPRokt
+
+- (void)selectPlacements:(NSString * _Nullable)identifier
+              attributes:(NSDictionary<NSString *, NSString *> * _Nullable)attributes {
+    NSDictionary<NSString *, NSString *> *mappedAttributes = [MPRokt mapUserAttributes:attributes mappingDictionary:@{}];
+    for (NSString *key in mappedAttributes) {
+        [[MParticle sharedInstance].identity.currentUser setUserAttribute:key value:mappedAttributes[key]];
+    }
+    
+    dispatch_async(dispatch_get_main_queue(), ^{
+        // Forwarding call to kits
+        MPForwardQueueParameters *queueParameters = [[MPForwardQueueParameters alloc] init];
+        [queueParameters addParameter:identifier];
+        [queueParameters addParameter:nil];
+        [queueParameters addParameter:nil];
+        [queueParameters addParameter:nil];
+        [queueParameters addParameter:nil];
+        [queueParameters addParameter:nil];
+        [queueParameters addParameter:nil];
+        [queueParameters addParameter:nil];
+        
+        SEL roktSelector = @selector(executeWithViewName:attributes:placements:onLoad:onUnLoad:onShouldShowLoadingIndicator:onShouldHideLoadingIndicator:onEmbeddedSizeChange:filteredUser:);
```

Comment:
maybe you can define this `executeWithViewName:attributes:placements:onLoad:onUnLoad:onShouldShowLoadingIndicator:onShouldHideLoadingIndicator:onEmbeddedSizeChange:filteredUser` somewhere since it's being used multiple times in the core sdk, as well as the kit?

Reply:
also, what would happen if they added a `placements`?  would that get set to nil?  all these parameters are optional besides identifier and attributes, and so they should be placed also. IMO we should code this so that a customer _can_ pass in arguments, but not necessarily all of them